app/src/main/java/org/gdg/frisbee/android/adapter/ContributorAdapter.java

```diff
@@ -15,46 +30,55 @@
 import org.gdg.frisbee.android.api.model.Contributor;
 import org.gdg.frisbee.android.app.App;
 
-/**
- * Created with IntelliJ IDEA.
- * User: maui
- * Date: 08.07.13
- * Time: 02:28
- * To change this template use File | Settings | File Templates.
- */
-public class ContributorAdapter extends ArrayAdapter<Contributor> {
+import butterknife.ButterKnife;
+import butterknife.InjectView;
 
-    private Context mContext;
-    private LayoutInflater mInflater;
+public class ContributorAdapter extends ArrayAdapter<Contributor> {
 
-    public ContributorAdapter(Context context, int textViewResourceId) {
-        super(context, textViewResourceId);
-        mContext = context;
-        mInflater = LayoutInflater.from(mContext);
+    public ContributorAdapter(Context context) {
+        super(context, R.layout.contributor_item);
     }
 
     @Override
-    public View getView(int position, View view, ViewGroup parent) {
-        if(view == null)
-            view = mInflater.inflate(R.layout.contributor_item, null);
+    public View getView(int position, View convertView, ViewGroup parent) {
+        if (convertView == null) {
+            convertView = View.inflate(getContext(), R.layout.contributor_item, null);
+            ViewHolder viewHolder = new ViewHolder(convertView);
+            convertView.setTag(viewHolder);
+        }
 
         Contributor contributor = getItem(position);
 
-        TextView name = (TextView)view.findViewById(R.id.contributorName);
-        name.setText(contributor.getLogin());
+        ViewHolder viewHolder = (ViewHolder) convertView.getTag();
+        viewHolder.contributorName.setText(contributor.getLogin());
+
         if (!TextUtils.isEmpty(contributor.getAvatarUrl())) {
-            final ImageView c = (ImageView) view.findViewById(R.id.contributorIcon);
             App.getInstance().getPicasso()
                     .load(contributor.getAvatarUrl())
-                    .into(c);
+                    .into(viewHolder.contributorAvatar);
         }
-        return view;
+
+        return convertView;
     }
 
     @Override
     public void addAll(Collection<? extends Contributor> contributors) {
+        setNotifyOnChange(false);
         for (Contributor c : contributors) {
             add(c);
         }
+        setNotifyOnChange(true);
+        notifyDataSetChanged();
```

Comment:
For devs knowing that setNotifyOnChange is set implicitly it is not clear
why it is mentioned explicitly. I recommend to remove it #prefmatters


  [image: --]
Friedger Müffke
[image: http://]about.me/friedger
<http://about.me/friedger>


2015-02-05 17:57 GMT+01:00 Said Tahsin Dane <notifications@github.com>:

> In
> app/src/main/java/org/gdg/frisbee/android/adapter/ContributorAdapter.java
> <https://github.com/gdg-x/frisbee/pull/172#discussion_r24178005>:
>
> >          for (Contributor c : contributors) {
> >              add(c);
> >          }
> > +        setNotifyOnChange(true);
> > +        notifyDataSetChanged();
>
> First of all thank you. :)
> I think that it is not that important. Thank you for the comment
> @ViliusKraujutis <https://github.com/ViliusKraujutis>
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/gdg-x/frisbee/pull/172/files#r24178005>.
>